lms/static/js/edxnotes/utils/notes_collector.js

```diff
@@ -0,0 +1,87 @@
+(function(define) {
+    'use strict';
+    define(['jquery', 'underscore', 'annotator_1.2.9'], function($, _, Annotator) {
+        var cleanup,
+            renderNotes,
+            fetchNotesWhenReady,
+            storeNotesRequestData,
+            searchRequestsData = [];
+
+        cleanup = function () {
+            searchRequestsData = [];
+        };
+
+        storeNotesRequestData = function (data) {
+            searchRequestsData.push(data);
+            fetchNotesWhenReady();
+        };
+
+        fetchNotesWhenReady = function () {
+            if ($('.edx-notes-wrapper').length != searchRequestsData.length) {
```

Comment:
I thought there were situations where we dynamically fetch new blocks and render them without reloading the page. Maybe that was just something that @ormsbee was prototyping though. I can imagine that even if we don't do this today, we will eventually need to support the option to dynamically switch the set of blocks that are shown, and thus we will need to clear the notes cache and request a new set.